lib/richmond/rtfm.rb

```diff
@@ -10,75 +10,77 @@ class RTFM
     include Richmond::Logging
 
     def initialize 
-      @record_pattern =/^\=begin richmond/i 
+      @record_pattern      =/^\=begin richmond/i
       @output_file_pattern = /output-file:\s+.*\s*/i
-      @end_record_pattern = /^\=end/i
-      @mode = :paused
-    end 
+      @end_record_pattern  = /^\=end/i
+      @mode                = :paused
+    end
 
     def parse_output_file(line)
-      match = line.match output_file_pattern
-      smatch = match
+      line.match(output_file_pattern)
         .to_s
         .gsub(/output-file:/, '')
         .strip
-   
-      smatch
     end
 
     def scan(dir)
-      logger.info "beginning scan"
-      files = Find.find(dir).to_a.reject!{|f| File.directory? f }
-      @output_filename = default_output_filename dir
-      result = scan_files files
-      logger.info "scan finished"
+      note "beginning scan"
+      mark_where_the_output_file_will_be_for dir
+      result = scan_files all_of_the_files_in(dir)
+      note "scan finished"
       result
     end
 
     def scan_files(files)
-      logger.info "scanning files: default output filename=#{@output_filename}"
+      note "scanning files: default output filename=#{@output_filename}"
       @result = ScanResult.new
+      files.each { |f| look_for_documentation_in f }
+      @result
+    end
+
+    def emit(input)
+      note "begin emitting files"
+      input.each_pair { |f, l| write_these_lines_to_a_file l, f }
+      note "finished emitting files"
+      #input.keys
+    end
+
+    private
 
-      files.each do |file|
-        @input_filename = File.expand_path file
-        lines = File.readlines file
-        lines.each do |line|
+    def look_for_documentation_in file
+      @the_name_of_the_file_being_read = File.expand_path file
+      File.readlines(file).each do |line|
+        begin
           stop_recording! line if stop_recording? line
           record! line if recording?
           start_recording! line if start_recording? line
+        rescue
+          @mode = :paused 
```

Comment:
Yeah, that was fixed on the feature/v1 branch. I don't know why that was happening. I added a line to sanitize each line as it came through to prevent this problem. I'm guessing it's because I've worked on this on both Windows and Mac, but the program should still work.